jenkinsbuilder/cccp_index_reader.py

```diff
@@ -327,9 +328,12 @@ def main(indexdlocation):
     )
     logger.info("List of stale projects: %s ", str(stale_projects))
 
-    # delete stale entries at jenkins
-    logger.debug("Deleting stale projects.")
-    delete_stale_projects_on_jenkins(stale_projects)
+    if not stale_projects:
+        # delete stale entries at jenkins
+        logger.debug("Deleting stale projects.")
+        delete_stale_projects_on_jenkins(stale_projects)
+    else:
+        logger.info("No stale projects in Jenkins.")
```

Comment:
`stale_projects` is supposed to be of `list` type. Taking an example:

```python
In [9]: l = [1, 2, 3]

In [10]: if not l:
    ...:     print "list has some entries"
    ...: else:
    ...:     print l
    ...:     
[1, 2, 3]

In [11]: l = []

In [12]: if not l:
    ...:     print "list has some entries"
    ...: else:
    ...:     print l
    ...:     
list has some entries
```

So, it looks to me like this logic is doing exactly opposite of what is expected from it. Am I looking at this properly? Feel free to call it the dumbest question. 😄

Reply:
ah well! my bad. fixing it. Thanks for pointing it out!